overlord/fdestate/backend/reseal.go

```diff
@@ -236,53 +235,43 @@ func doReseal(manager FDEStateManager, method device.SealingMethod, rootdir stri
 		}
 	}
 
-	switch method {
-	case device.SealingMethodFDESetupHook:
-		// Note this is the fallback key path for old installations. The file might not exist.
-		primaryKeyFile := filepath.Join(boot.InstallHostFDESaveDir, "aux-key")
-		// All devices should have the same primary key
-		primaryKeyGetter := func() ([]byte, error) {
-			return secbootGetPrimaryKey(devices, primaryKeyFile)
+	primaryKeyGetter := func() ([]byte, error) {
+		saveFDEDir := dirs.SnapFDEDirUnderSave(dirs.SnapSaveDirUnder(rootdir))
+		fallbackPrimaryKeyFiles := []string{
+			filepath.Join(saveFDEDir, "aux-key"),
+			filepath.Join(saveFDEDir, "tpm-policy-auth-key"),
 		}
-		for _, key := range keys {
-			if err := secbootResealKeysWithFDESetupHook([]secboot.KeyDataLocation{key.location}, primaryKeyGetter, key.params.Models, key.params.BootModes); err != nil {
-				return err
-			}
+		return secbootGetPrimaryKey(devices, fallbackPrimaryKeyFiles)
+	}
+
+	var allResealedKeys secboot.UpdatedKeys
+	for _, key := range keys {
+		params := &secboot.ResealKeyParams{
+			GetPrimaryKey:              primaryKeyGetter,
```

Comment:
this is slightly odd now, shouldn't we pass now the Devices and the FallbackPrimaryKeyFiles instead?

Reply:
Not sure how that will work with https://github.com/canonical/snapd/pull/15694

Reply:
I changed it. But the other PR might need to introduce a call back again.